### ISSUE TITLE
How to set-up vertex?

### ISSUE THREAD

**Issue by adriangalilea** [Aider-AI/aider#993]
### Issue

I had installed all it asked me to, and set the variables it requested, still complains about needing to install:
```
BadRequestError: litellm.BadRequestError: VertexAIException BadRequestError - vertexai import failed 
please run pip install -U google-cloud-aiplatform "anthropic[vertex]"
```

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Did you do the pip install that the error specified?

**Comment by adriangalilea** [comment]
> Thanks for trying aider and filing this issue.
> 
> Did you do the pip install that the error specified?

Yup,  and installed the google cloud cli in order to authenticate and so on, and nothing, no idea.

**Comment by paul-gauthier** [comment]
Did you install aider with `pip` or `pipx`? You'd need to install the google packages differently if you installed aider with `pipx`.

**Comment by adriangalilea** [comment]
> Did you install aider with `pip` or `pipx`? You'd need to install the google packages differently if you installed aider with `pipx`.

I did install with `pipx` how shall I do it?

**Comment by sholub89** [comment]
@adriangalilea , also after doing the above it is sometimes required to close the current terminal window and open a new one so the changes will apply. Try that and I hope this will work for you

**Comment by adriangalilea** [comment]
> @adriangalilea , also after doing the above it is sometimes required to close the current terminal window and open a new one so the changes will apply. Try that and I hope this will work for you

Will test out in a couple weeks, thanks.

**Comment by msirprism** [comment]
Here is the command for pipx : pipx runpip aider-chat install -U google-cloud-aiplatform "anthropic[vertex]"

**Comment by blainehansen** [comment]
For posterity, if your `aider` install is installed with `uv` (you can check with `cat $(which aider)`) you need to use `uv`'s [additional tool dependency system using `--with`](https://docs.astral.sh/uv/concepts/tools/#including-additional-dependencies):

```
uv tool install --force --python python3.12 --with google-cloud-aiplatform --with "anthropic[vertex]" aider-chat@latest
```
